src/epic/web/index.js

```diff
@@ -0,0 +1,418 @@
+import { getIframeId, getWebfonts, onViewport, resizeIframeHeight } from '../../_shared/js/messages';
+
+// Expects all data required by the javascript in this file
+// to be included as data attributes on an element with class name js-dfp-data.
+// To get the value of the attribute e.g. data-referrer-url the name parameter should be referrer-url.
+function getDataAttribute(name) {
+    try {
+        return document.getElementsByClassName('js-dfp-data')[0].getAttribute('data-' + name)
+    } catch (_) {
+        return;
+    }
+}
+
+// This function gets the referrer url from a data attribute on the Epic button,
+// percent encodes the url, and adds it to the acquisitionData GET parameter (acquisition data).
+//
+// Why is this needed?
+// -------------------
+// The url of the referring page can be obtained from the %%PATTERN:url%% DFP macro,
+// however, this is not percent encoded, and there are known instances where it being
+// unescaped prevents the acquisition data being deserialised by support frontend
+// e.g. if the referrer url contains a fragment identifier (#)
+//
+// Why not use the [%URI_ENCODE:variable%] DFP macro?
+// --------------------------------------------------
+// It only works with variables that are defined in the Native Ad Format; url is not one of these.
+//
+// Why include the unescaped referrer url as a data attribute and not in the acquisition data?
+// -------------------------------------------------------------------------------------------
+// If it is included as an unescaped field in the acquisition data and this function call fails,
+// then potentially support frontend would not be able to deserialise the acquisition data
+// (see previous question for more details).
+function addEncodedReferrerUrlToClickThroughLink() {
+
+    const buttons = document.getElementsByClassName('js-epic-single-button');
```

Comment:
`.querySelector('.js-epic-single-button')` could be neater - it will return `null` or the first matching

Reply:
tres nice